### PR TITLE
Add SVI MAC address option in mso_schema_site_bd

### PR DESCRIPTION
**Pull request by xinyuezhao**
- Add SVI MAC address in mso_schema_site_bd
- Update test file for mso_schema_site_bd to test SVI MAC address
- Remove space in template

### CODE REVIEW

**Review by shrsr — COMMENTED** [review]
LGTM

File: plugins/modules/mso_dhcp_relay_policy_provider.py [hunk 1]
```diff
@@ -158,6 +158,8 @@ def main():
     tenant = module.params.get("tenant")
     schema = module.params.get("schema")
     template = module.params.get("template")
+    if template is not None:
+        template = template.replace(' ', '')
```
Comment (lhercot) [inline plugins/modules/mso_dhcp_relay_policy_provider.py:162]: I would put the replace on line 160 like the other occurrences

File: plugins/modules/mso_schema_site.py [hunk 2]
```diff
@@ -123,6 +123,8 @@ def main():
     schema = module.params.get('schema')
     site = module.params.get('site')
     template = module.params.get('template')
+    if template is not None:
+        template = template.replace(' ', '')
```
Comment (lhercot) [inline plugins/modules/mso_schema_site.py:127]: I would put the replace on line 160 like the other occurrences

File: plugins/modules/mso_schema_template.py [hunk 3]
```diff
@@ -130,6 +130,8 @@ def main():
     tenant = module.params.get('tenant')
     schema = module.params.get('schema')
     template = module.params.get('template')
+    if template is not None:
+        template = template.replace(' ', '')
```
Comment (lhercot) [inline plugins/modules/mso_schema_template.py:134]: I would put the replace on line 160 like the other occurrences

File: plugins/modules/mso_schema_template_contract_filter.py [hunk 4]
```diff
@@ -187,14 +187,16 @@ def main():
...
+    template = module.params.get('template').replace(' ', '')
     contract = module.params.get('contract')
     contract_display_name = module.params.get('contract_display_name')
     contract_filter_type = module.params.get('contract_filter_type')
     contract_scope = module.params.get('contract_scope')
     filter_name = module.params.get('filter')
     filter_directives = module.params.get('filter_directives')
     filter_template = module.params.get('filter_template')
+    if filter_template is not None:
+        filter_template = filter_template.replace(' ', '')
```
Comment (lhercot) [inline plugins/modules/mso_schema_template_contract_filter.py:199]: I would put the replace on line 197 like the other occurrences

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml [hunk 5]
```diff
@@ -354,9 +354,6 @@
     epg: ansible_test_1
     selector: selector_2
     expressions:
-      - type: expression_1
-        operator: in
-        value: test
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:359]: Not sure why we removed this block. I understand the type should be different but we should still have a case with more than 1 expression in one go.

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml [hunk 6]
```diff
@@ -382,7 +379,7 @@
     that:
     - nm_change_selector_2 is changed
     - nm_change_selector_2.current.name == "selector_2"
-    - nm_change_selector_2.current.expressions | length == 7
+    - nm_change_selector_2.current.expressions | length == 6
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:382]: If we add back the expression as asked, this should go back up to 7.

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml [hunk 7]
```diff
@@ -393,9 +390,6 @@
     epg: ansible_test_1
     selector: selector_2
     expressions:
-      - type: expression_1
-        operator: in
-        value: test
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:398]: Same here

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml [hunk 8]
```diff
@@ -421,7 +415,7 @@
     that:
     - nm_change_selector_2_again is not changed
     - nm_change_selector_2_again.current.name == "selector_2"
-    - nm_change_selector_2_again.current.expressions | length == 7
+    - nm_change_selector_2_again.current.expressions | length == 6
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:418]: Same here

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml [hunk 9]
```diff
@@ -442,7 +436,7 @@
     - cm_query_all_selectors.current[0].name == "selector_1"
     - cm_query_all_selectors.current[1].name == "selector_2"
     - cm_query_all_selectors.current[0].expressions == []
-    - cm_query_all_selectors.current[1].expressions | length == 7
+    - cm_query_all_selectors.current[1].expressions | length == 6
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:439]: Same here

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml [hunk 10]
```diff
@@ -462,7 +456,7 @@
     - nm_query_all_selectors.current[0].name == "selector_1"
     - nm_query_all_selectors.current[1].name == "selector_2"
     - nm_query_all_selectors.current[0].expressions == []
-    - nm_query_all_selectors.current[1].expressions | length == 7
+    - nm_query_all_selectors.current[1].expressions | length == 6
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:459]: Same here

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml [hunk 11]
```diff
@@ -517,27 +511,24 @@
     that:
     - nm_query_selector2 is not changed
     - nm_query_selector2.current.name == "selector_2"
-    - nm_query_selector2.current.expressions | length == 7
+    - nm_query_selector2.current.expressions | length == 6
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:514]: Same here

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml [hunk 12]
```diff
@@ -517,27 +511,24 @@
...
+    - nm_query_selector2.current.expressions[3].operator == "notEquals"
     - nm_query_selector2.current.expressions[3].value == "test"
-    - nm_query_selector2.current.expressions[4].key == "Custom:expression_4"
-    - nm_query_selector2.current.expressions[4].operator == "notEquals"
-    - nm_query_selector2.current.expressions[4].value == "test"
-    - nm_query_selector2.current.expressions[5].key == "Custom:expression_5"
-    - nm_query_selector2.current.expressions[5].operator == "keyExist"
-    - nm_query_selector2.current.expressions[5].value == ""
-    - nm_query_selector2.current.expressions[6].key == "Custom:expression_6"
-    - nm_query_selector2.current.expressions[6].operator == "keyNotExist"
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:540]: If you add back expression_7, change this part to comply

File: tests/integration/targets/mso_schema_template_deploy/tasks/main.yml [hunk 13]
```diff
@@ -180,5 +186,6 @@
   assert:
     that:
     - item is not changed
-    - item.msg == "Successfully deployed"
-  loop: "{{ undeploy_template.results }}"
```
Comment (lhercot) [inline tests/integration/targets/mso_schema_template_deploy/tasks/main.yml:184]: We should still test the "Successfully deployed" on the versions under 3.1

**Review by lhercot — CHANGES_REQUESTED** [review]
Let me know if any of those changes are not possible.

File: plugins/modules/mso_dhcp_relay_policy_provider.py (hunk 1 above)
Comment (xinyuezhao) [reply plugins/modules/mso_dhcp_relay_policy_provider.py:162]: The template is not required, so we need to check whether the template is none before replacing.

File: plugins/modules/mso_schema_site.py (hunk 2 above)
Comment (xinyuezhao) [reply plugins/modules/mso_schema_site.py:127]: The template is not required, so we need to check whether the template is none before replacing.

File: plugins/modules/mso_schema_template.py (hunk 3 above)
Comment (xinyuezhao) [reply plugins/modules/mso_schema_template.py:134]: The template is not required, so we need to check whether the template is none before replacing.

File: plugins/modules/mso_schema_template_contract_filter.py (hunk 4 above)
Comment (xinyuezhao) [reply plugins/modules/mso_schema_template_contract_filter.py:199]: The template of filter is not required, so we need to check whether the template is none before replacing.

File: tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml (hunk 7 above)
Comment (xinyuezhao) [reply tests/integration/targets/mso_schema_template_anp_epg_selector/tasks/main.yml:398]: There are already 6 expressions and we don't need 7 expressions any more

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**